### ISSUE TITLE
[Flaky test] multibatch result for pg_join

### ISSUE THREAD

**Issue by mkindahl** [timescale/timescaledb#5167]
### Which test is flaky?

pg_join

### Since when has the test been flaky?

January 10, 2023

### Link to the failed test run

https://github.com/timescale/timescaledb/actions/runs/3874122170/jobs/6605057165

### Log output

```bash
diff -u /home/runner/work/timescaledb/timescaledb/test/expected/pg_join.out /home/runner/work/timescaledb/timescaledb/build/test/results/pg_join.out
--- /home/runner/work/timescaledb/timescaledb/test/expected/pg_join.out	2023-01-09 13:29:51.102144268 +0000
+++ /home/runner/work/timescaledb/timescaledb/build/test/results/pg_join.out	2023-01-09 13:39:33.722273643 +0000
@@ -199,3 +199,12 @@
  (1 row)
  
   table_name 
+@@ -5525,7 +5525,7 @@
+ 
+  multibatch 
+ ------------
+- f
++ t
+ (1 row)
+ 
+  count
```

**Comment by akuzm** [comment]
```
ci_stats=> select count(*) from test where test_name = 'pg_join' and test_status != 'ok' and job_date > now() - interval '1 week';
 count 
───────
     4
(1 row)
```

This is concerning, a pretty basic test started to fail. I think we should investigate this before releasing a new version.

**Comment by akuzm** [comment]
```
ci_stats=> select distinct log_contents from test natural join log where test_name = 'pg_join' and test_status != 'ok' and job_date > now() - interval '1 week';
log_contents|c 0001: @@ -199,3 +199,12 @@
p 0002:   (1 row)
p 0003:   
b 0004:    table_name 
c 0005: +@@ -5525,7 +5525,7 @@
p 0006: + 
p 0007: +  multibatch 
b 0008: + ------------
b 0009: +- f
b 0010: ++ t
b 0011: + (1 row)
b 0012: + 
b 0013: +  count 

log_contents|c 0001: @@ -199,3 +199,12 @@
p 0002:   (1 row)
p 0003:   
b 0004:    table_name 
c 0005: +@@ -5525,7 +5525,7 @@
p 0006: + 
p 0007: +  multibatch 
b 0008: + ------------
b 0009: +- t
b 0010: ++ f
b 0011: + (1 row)
b 0012: + 
b 0013: +  count 
```
Funny that it has failures with exactly opposite reference in the same row. Probably because the reference is also generated by the test, and it's also unstable.

**Comment by mkindahl** [comment]
https://github.com/timescale/timescaledb/actions/runs/4210024942/jobs/7307415667

**Comment by mkindahl** [comment]
https://github.com/timescale/timescaledb/actions/runs/4210024942/jobs/7307415724

**Comment by mkindahl** [comment]
https://github.com/timescale/timescaledb/actions/runs/4223209693/jobs/7332676662

**Comment by jnidzwetzki** [comment]
Here we have a new one: https://github.com/timescale/timescaledb/actions/runs/4379943887/jobs/7666444168

**Comment by mkindahl** [comment]
https://github.com/timescale/timescaledb/actions/runs/4806124707/jobs/8554607109?pr=5615

**Comment by jnidzwetzki** [comment]
Failed again in CI: https://github.com/timescale/timescaledb/actions/runs/5284753117/jobs/9562506673

**Comment by mkindahl** [comment]
https://github.com/timescale/timescaledb/actions/runs/5962846025/job/16174859787?pr=5994

**Comment by jnidzwetzki** [comment]
It might be fixed by #6088. So, closing the issue for now.
